agent.go

```diff
@@ -385,6 +385,100 @@ func (c *controller) agentClose() {
 	agent.networkDB.Close()
 }
 
+type Task struct {
+	Name       string
+	EndpointID string
+	EndpointIP string
+	Info       string
+}
+
+type ServiceInfo struct {
+	VIP   string
+	Tasks []Task
+	Ports []string
+}
+type epRecord struct {
+	ep   EndpointRecord
+	info string
+}
+
+func (n *network) Services() map[string]ServiceInfo {
+	eps := make(map[string]epRecord)
+
+	if !n.isClusterEligible() {
+		return nil
+	}
+	agent := n.getController().getAgent()
+	if agent == nil {
+		return nil
+	}
+
+	// Walk through the libnetwork's "endpoint_table" and fetch the driver
+	// agnostic endpoint info
+	entries := agent.networkDB.GetTableByNetwork("endpoint_table", n.id)
+	for eid, value := range entries {
+		var epRec EndpointRecord
+		if err := proto.Unmarshal(value.([]byte), &epRec); err != nil {
+			logrus.Errorf("Unmarshal of %s failed", eid)
```

Comment:
Please add some context to this log
`logrus.Errorf("Unmarshal of endpoint record for %s failed, while retrieving services info: %v", eid, err)`